src/configureWorkspace/configurePython.ts

```diff
@@ -3,68 +3,224 @@
+
+    return dockerComposeDebugfile
+        .replace(/\$service_name\$/g, serviceName)
+        .replace(/\$dbg_volume\$/g, `- ${PythonExtensionHelper.getLauncherFolderPath()}:/pydbg`)
+        .replace(/\$entrypoint\$/g, entrypoint)
+        .replace(/\$ports\$/g, getComposePorts(ports, PythonDefaultDebugPort));
+}
+
+function genRequirementsFile(projectType: PythonProjectType): string {
+    let contents = '# Add requirements when needed'
+
+    switch (projectType) {
+        case "django":
+            contents = djangoRequirements;
+            break;
+        case "flask":
+            contents = flaskRequirements
+            break;
+        default:
+    }
+
+    return contents;
+}
+
+async function initializeForDebugging(context: ScaffolderContext, dockerfile: string, ports: number[], generateComposeFiles: boolean,
+                                      target: PythonFileTarget | PythonModuleTarget, projectType: PythonProjectType): Promise<void> {
+    const scaffoldContext: DockerDebugScaffoldContext = {
+        folder: context.folder,
+        platform: "python",
+        actionContext: context,
+        dockerfile: dockerfile,
+        generateComposeTask: generateComposeFiles,
+        ports: ports
+    }
+
+    const pyOptions: PythonScaffoldingOptions = {
+        target: target,
+        projectType: projectType
+    }
+
+    await dockerDebugScaffoldingProvider.initializePythonForDebugging(scaffoldContext, pyOptions);
+}
+
+export async function promptForLaunchFile(projectType?: PythonProjectType) : Promise<PythonFileTarget | PythonModuleTarget> {
+    const launchFilePrompt = defaultLaunchFile.get(projectType);
+
+    const opt: vscode.InputBoxOptions = {
+        placeHolder: launchFilePrompt.defaultFile,
+        prompt: launchFilePrompt.prompt,
+        value: launchFilePrompt.defaultFile,
+        validateInput: (value: string): string | undefined => { return value && value.trim().length > 0 ? undefined : "Enter a valid Python file path/module." }
+    };
+
+    const file = await ext.ui.showInputBox(opt);
+
+    // If the input has the .py extension, then assume it is a file.
+    // TODO: is there a more robust way to check?
+    if (file.toLowerCase().endsWith(".py")) {
+        return { file: file.replace(/\\/g, '/') };
+    } else {
+        return { module: file};
+    }
+}
+
+export async function scaffoldPython(context: ScaffolderContext): Promise<ScaffoldFile[]> {
+    const properties: TelemetryProperties & ConfigureTelemetryProperties = context.telemetry.properties;
+    const serviceName = context.folder.name;
+    const rootFolderPath: string = context.rootFolder;
+    const outputFolder = context.outputFolder ?? rootFolderPath;
+
+    const generateComposeFiles = await context.captureStep("compose", quickPickGenerateComposeFiles)();
+    const projectType = getPythonProjectType(context.platform);
+
+    const defaultPort = PythonDefaultPorts.get(projectType);
+    let ports = [];
+
+    if (defaultPort) {
```

Comment:
We only prompt the user if there's a default port for a project type?  That is, we don't prompt for project types that don't have a well known port?  Isn't that exactly when we *should* prompt?

Reply:
The thinking is that for projects under the "general" category, we shouldn't assume a port has to be exposed (i.e. it could be a console-like app). While for django and flask, there has to be a port since it's a web app essentially.

Initially, I had it prompting for a port all the time, but based on one of the feedback meetings there was a vote to not do that for "general" and the user can easily expose a port if needed.

Reply:
Ok, I'll buy that.  We can see if we get any feedback from users on that.